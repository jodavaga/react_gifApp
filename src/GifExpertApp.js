import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return(
        <>
            <h1>Gif Expert App</h1>
            <AddCategory 
                setCategories = {setCategories} 
            />
            <hr />

            <ol>
                {
                    categories.map(cat => (
                        <GifGrid 
                            key={ cat } 
                            category={ cat }
                        />
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpertApp;