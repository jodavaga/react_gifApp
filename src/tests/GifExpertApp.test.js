import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('<GifExpertApp /> tests', () => {

    test('should render component as expected', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should render component as with amount of <GifGrid />', () => {
        
        const categories = ['rick', 'morty'];
        const wrapper = shallow(<GifExpertApp myCategories={ categories } />);
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
})