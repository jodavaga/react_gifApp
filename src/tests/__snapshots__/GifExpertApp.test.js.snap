// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> tests should render component as expected 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
